sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/data/redis/AzureLettucePasswordlessAutoConfiguration.java

```diff
@@ -54,8 +72,12 @@ AzureRedisPasswordlessProperties redisPasswordlessProperties() {
     AzureRedisCredentials azureRedisCredentials(RedisProperties redisProperties,
                                                 AzureRedisPasswordlessProperties azureRedisPasswordlessProperties,
                                                 AzureGlobalProperties azureGlobalProperties) {
+        Properties properties = azureRedisPasswordlessProperties.toPasswordlessProperties();
+        enhancePasswordlessProperties(applicationContext, "spring.data.redis.azure",
+            azureRedisPasswordlessProperties, properties);
+        TokenCredentialProvider provider = TokenCredentialProviders.createInstance(new TokenCredentialProviderOptions(properties));
```

Comment:
I don't think we should do it this way, we should only pass in the properties, and let the provider chosen by the azure-identity extension to resolve the bean based on the properties/options